lib/controllers/postGrams.js

```diff
@@ -22,7 +22,26 @@ module.exports = Router()
       .findById(req.params.id)
       .then(gram => res.send(gram))
       .catch(next);
-  });
+  })
+
+  .patch('/:id', ensureAuth, (req, res, next) => {
+    PostGram
+      .update(req.params.id, { ...req.body, userId: req.user.id })
+      .then(gram => res.send(gram))
+      .catch(next);
+  })
   
+  .delete('/:id', ensureAuth, (req, res, next) => {
+    PostGram
+      .delete(req.params.id, { userId: req.user.id })
+      .then(gram => res.send(gram))
+      .catch(next);
+  })
 
+  .get('/prolific', ensureAuth, (req, res, next) => {
+    PostGram
+      .prolific()
```

Comment:
I believe you guys meant to use your findPopular() method from your Postgram model here, instead of prolific, which is from your User model.

Reply:
so route from the rubric for reference should be ```GET /posts/popular```